src/registerAllocator/RegisterAllocator.class.ts

```diff
@@ -593,7 +595,7 @@ export class RegisterAllocator {
       }
     }
 
-    if (caf(AllocationFlags.ONE_IN_MUST_BE_IN_RDX) && !inAllocations.includes(Register.rdx)) {
+    if (caf(AllocationFlags.ONE_IN_MUST_BE_IN_RDX) && !allocationReq.in.some((i) => this._allocations[i].store === Register.rdx)) {
```

Comment:
Why do we check`this._allocations[i]` instead of `inAllocations`?

Reply:
I do not remember for sure, but I think it was a difference between `[rax+4]` and `arg1[1]`. You're right to question this code as I was working from examples, not deep understanding.

Reply:
Thing is, I could see this working but also failing. `inAllocations` is being used and modified in this function, whereas `_allocations` is modified by the `getW()` function if I remember correctly. I'd want to not touch anything if we don't know if its necessary. This whole RegisterAllocator needs a rewrite, but I've never had the time.